pkg/upgrade/control_plane.go

```diff
@@ -890,7 +904,10 @@ func (u *ControlPlaneUpgrader) updateBootstrapConfig(replacementKey ctrlclient.O
 	// modify bootstrap config
 	bootstrap.SetName(replacementKey.Name)
 	bootstrap.SetResourceVersion("")
-	bootstrap.SetOwnerReferences(nil)
+
+	// Remove only the Machine owner reference so we preserve other refs
+	ownerRefs := getAllButMachineOwners(bootstrap.GetOwnerReferences())
+	bootstrap.SetOwnerReferences(ownerRefs)
```

Comment:
Is there a test for this change?

Reply:
Testing this change seems best suited for the integration test suite, which is currently broken. I've gone ahead and added a unit test for the filtering mechanism, though.

Reply:
I take it back: there are no unit test for `updateBootstrapConfig()` right now, so you can't update any existing unit test :(

Maybe let's add one for `getAllButMachineOwners()`, at least.